ClientApp/KachnaOnline/src/app/shared/services/authentication.service.ts

```diff
@@ -64,22 +64,30 @@ export class AuthenticationService {
     this.http.get<AccessTokens>(`${AUTH_API}/accessTokenFromSession`, { params: params }).toPromise()
         .then((res) => {
           this.handleAccessTokens(res);
-
-          this.refreshLocalTokenIntervalHandle = setInterval( () => {
-              this.refreshAuthToken();
-            },
-            2500 * 1000); // FIXME: Refresh according to expire value returned from KO API.
         }).catch((error: any) => {
           this.toastr.error("Načtení přístupových údajů selhalo.", "Autentizace");
           return throwError(error);
     });
   }
 
+  private setRefreshIntervalForAuthToken() {
+    clearInterval(this.refreshLocalTokenIntervalHandle);
+    this.refreshLocalTokenIntervalHandle = setInterval(() => {
+        this.refreshAuthToken();
+      },
+      this.countLocalAccessTokenRefreshIntervalTime()); // TODO
```

Comment:
Why the TODO? As far as I can see, you are already using the expiration so all should be fine, isn't it?

Reply:
Sorry, last changes not pushed yet.